odxtools/basevariantmatcher.py

```diff
@@ -0,0 +1,35 @@
+# SPDX-License-Identifier: MIT
+from typing import List
+
+from .diaglayers.basevariant import BaseVariant
+from .variantmatcher import VariantMatcher
+
+
+class BaseVariantMatcher(VariantMatcher):
+    """BaseVariantMatcher implements the matching algorithm of base variants using
+    BASE-VARIANT-PATTERNs according to ISO 22901-1.
+
+    Usage (example):
+
+    ```python
+
+    # initialize the matcher with a list of base variants
+    matcher = BaseVariantMatcher(base_variant_candidates=[...], use_cache=use_cache)
+
+    # run the request loop to obtain responses for every request
+    for use_physical_addressing, encoded_request in matcher.request_loop():
+        if use_physical_addressing:
+            resp = send_to_ecu_using_physical_addressing(encoded_request)
+        else:
+            resp = send_to_ecu_using_functional_addressing(encoded_request)
+        matcher.evaluate(resp)
+
+    # result
+    if matcher.has_match()
+        matching_bv = matcher.matching_variant
+        assert isinstance(matching_bv, BaseVariant)
+    ```
+    """
+
+    def __init__(self, base_variant_candidates: List[BaseVariant], use_cache: bool = True):
```

Comment:
Really need the prefix?
```suggestion
    def __init__(self, candidates: List[BaseVariant], use_cache: bool = True):
```

Reply:
fixed: https://github.com/mercedes-benz/odxtools/pull/394/commits/98d90caa4efd9f14603fad9175da279406da0b89